NetworkingSampleApp/NetworkingSampleApp/Scenes/Download/DownloadProgressViewModel.swift

```diff
@@ -9,28 +9,43 @@ import Foundation
 import Networking
 
 @MainActor
-final class DownloadProgressViewModel: ObservableObject {
+final class DownloadProgressViewModel: TaskProgressViewModel {
     private let task: URLSessionTask
     
-    @Published var state: DownloadProgressState = .init()
-    
+    let isRetryable = false
+    private(set) var title: String = ""
```

Comment:
Why not have a @Published private(set) var state: DownloadState instead?

Reply:
I reworked it to conform to the `TaskProgressViewModel` protocol.